### PR TITLE
[#2837] Publish notifications from the Device Registry with Kafka.

### PR DESCRIPTION
**Pull request by b-abel**
This is for #2837. It adds notification support to the device registry using the Kafka client.

### CODE REVIEW

**Comment by b-abel** [comment]
The Kafka producers and consumers for the notifcations use `JSON` as the type of the record values. The class `KafkaProducerConfigProperties` on the other hand expects the value to be of type `Buffer`. The difference between both is not so much technical, rather the APIs are different. We could change the type for notifications to `Buffer`, but this becomes obsolete with #2904 when we use different configuration properties. I tend to defer this PR until #2904 is resolved. @calohmn @kaniyan Are you planning to work on #2904?

File: services/device-registry-base/src/test/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    /**
+     * Verifies that
+     * {@link AbstractCredentialsManagementService#updateCredentials(String, String, List, Optional, Span)} publishes
+     * the expected notification.
+     *
+     * @param context The vert.x test context.
+     */
+    @Test
+    public void testNotificationOnUpdateDevice(final VertxTestContext context) {
```
Comment (sophokles73) [inline services/device-registry-base/src/test/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementServiceTest.java:120]: onUpdateCredentials?

**Comment by sophokles73** [comment]
@b-abel can you rebase? I have already incorporated the fix for calculating the command sending timeout in the ITs into master ...

**Comment by b-abel** [comment]
> @b-abel can you rebase? I have already incorporated the fix for calculating the command sending timeout in the ITs into master ...

Done.

File: services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractDeviceManagementService.java [hunk 2]
```diff
@@ -372,6 +406,7 @@ protected String generateDeviceId(final String tenantId) {
                     }
                     return processDeleteDevicesOfTenant(tenantId, span);
                 })
+                // TODO send notification
```
Comment (sophokles73) [inline services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractDeviceManagementService.java:409]: are you going to address this as part of this PR or will you create another PR for it?

Comment (b-abel) [reply services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractDeviceManagementService.java:409]: TBH I haven't thought much about this yet and have no idea how to address this. So, yes this should be part of another PR.

Comment (sophokles73) [reply services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractDeviceManagementService.java:409]: Ok, then please don't forget to do so ;-) I can imagine using a dedicated notification type for that purpose.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**